core/src/main/java/org/mule/TransformationService.java

```diff
@@ -0,0 +1,513 @@
+        {
+            message.setPayload(getPayload(message, DataTypeFactory.create(outputType)));
+        }
+    }
+
+
+    /**
+     * Attempts to obtain the payload of this message with the desired Class type. This will
+     * try and resolve a transformer that can do this transformation. If a transformer cannot be found
+     * an exception is thrown.  Any transformers added to the registry will be checked for compatibility.
+     *
+     * If the existing payload is consumable (i.e. can't be read twice) then the existing payload of the message will be
+     * replaced with a byte[] representation as part of this operations.
+     *
+     * @param outputType the desired return type
+     * @return The converted payload of this message. Note that this method will not alter the payload of this
+     * message *unless* the payload is an InputStream in which case the stream will be read and the payload will become
+     * the fully read stream.
+     * @throws TransformerException if a transformer cannot be found or there is an error during transformation of the
+     * payload
+     */
+    public Object getPayload(MuleMessage message, Class outputType) throws TransformerException
+    {
+        return getPayload(message, DataTypeFactory.create(outputType));
+    }
+
+    /**
+     * Attempts to obtain the payload of this message with the desired Class type. This will
+     * try and resolve a transformer that can do this transformation. If a transformer cannot be found
+     * an exception is thrown.  Any transformers added to the registry will be checked for compatibility
+     *
+     * If the existing payload is consumable (i.e. can't be read twice) then the existing payload of the message will be
+     * replaced with a byte[] representation as part of this operations.
+     *
+     * @param outputType the desired return type
+     * @return The converted payload of this message. Note that this method will not alter the payload of this
+     * message *unless* the payload is an InputStream in which case the stream will be read and the payload will become
+     * the fully read stream.
+     * @throws TransformerException if a transformer cannot be found or there is an error during transformation of the
+     * payload
+     */
+    public <T> Object getPayload(MuleMessage message, DataType<T> outputType) throws TransformerException
+    {
+        if (message instanceof MuleMessageCollection)
+        {
+            MuleMessageCollection messageCollection = (MuleMessageCollection) message;
+            if (messageCollection.isInvalidatedPayload())
+            {
+                return getPayload(messageCollection, outputType);
+            }
+            else
+            {
+                List<T> results = new ArrayList<T>(messageCollection.getMessageList().size());
+                for (MuleMessage messageInList : messageCollection.getMessageList())
```

Comment:
I think you mean streams, I could but the fact is that it's not possible when map() operations throw checked exceptions.